remotecommanddispatcher/src/main/java/com/tealium/remotecommanddispatcher/RemoteCommandParser.kt

```diff
@@ -65,8 +76,86 @@ class RemoteCommandParser {
+        private fun checkAndSplitDestinationList(
+            lookupValue: String,
+            delimiter: String
+        ): List<String> {
+            return lookupValue.split(delimiter).map { it.trim() }
+        }
+
+        fun processStaticMappings(
+            statics: Map<String, Any>?,
+            payload: Map<String, Any>,
+            delimiters: Delimiters
+        ): Map<String, Any> {
+            val processedPayload = payload.toMutableMap()
+            statics?.let { staticMap ->
+                staticMap.forEach { (key, value) ->
+                    val splitKeys = splitCompoundKeys(key, delimiters)
+                    if (matchKeys(splitKeys, payload)) {
+                        val valueMap =
+                            (value as Map<*, *>).entries.associate { entry -> entry.key.toString() to entry.value as Any }
+                        processedPayload.putAll(valueMap)
+                    }
+                }
+            }
+            return processedPayload.toMap()
+        }
+
+        fun extractCommandNames(
+            commands: Map<String, String>?,
+            payload: Map<String, Any>,
+            delimiters: Delimiters
+        ): String {
+            val eventType = payload[Dispatch.Keys.TEALIUM_EVENT_TYPE] as? String
+            val commandsList = mutableListOf<String>()
+
+            commands?.let { commandsMap ->
+                commandsMap.forEach { (key, value) ->
+                    val splitCommands = splitCompoundKeys(key, delimiters)
+                    if (matchKeys(splitCommands, payload)) {
+                        commandsList.addAll(checkAndSplitDestinationList(value, ","))
+                    }
+                }
+            }
+
+            commands?.get(Settings.ALL_EVENTS)?.let {
+                if (eventType == DispatchType.EVENT) commandsList.add(it)
+            }
+            commands?.get(Settings.ALL_VIEWS)?.let {
+                if (eventType == DispatchType.VIEW) commandsList.add(it)
+            }
+
+            return commandsList.joinToString(",")
+        }
+
+        private fun splitCompoundKeys(key: String, delimiters: Delimiters): Map<String, String> {
+            val result = mutableMapOf<String, String>()
+            val keys = checkAndSplitDestinationList(key, delimiters.keysSeparationDelimiter)
+            keys.forEach { k ->
+                val keyValue = checkAndSplitDestinationList(k, delimiters.keysEqualityDelimiter)
+                if (keyValue.size == 1) {
+                    result[Key.TEALIUM_EVENT] = keyValue.first()
+                } else {
+                    result[keyValue.first()] = keyValue.last()
+                }
+            }
+            return result.toMap()
+        }
+
+        private fun matchKeys(
+            keyValues: Map<String, String>,
+            payload: Map<String, Any>
+        ): Boolean {
+            val result = keyValues.filter { (key, value) ->
+                if (!payload.containsKey(key)) {
```

Comment:
There's a couple of issues to address in this method.
1. we can short-circuit the `return false` as soon as we have a non-matching pair.
2. Line 154 needs to be able to handle some basic type comparison handling - i.e. if a payload contains an integer, then it needs to pass equality with the string that would have been brought in from the JSON. It may be that a `toString()` on the payload value is sufficient to cover Integers/Doubles, Strings and Booleans which are the most likely rule values.